core/app/views/spree/order_mailer/confirm_email.html.erb

```diff
@@ -8,70 +8,56 @@
-              </tr>
-            <% end %>
-          <% end %>
-        <% end %>
-        <% @order.shipments.group_by { |s| s.selected_shipping_rate.try(:name) }.each do |name, shipments| %>
-          <tr>
-            <td></td>
-            <td><%= Spree.t(:shipping) %> <%= name %>:</td>
-            <td><%= Spree::Money.new(shipments.sum(&:discounted_cost), currency: @order.currency) %></td>
-          </tr>
-        <% end %>
-        <% if @order.all_adjustments.eligible.tax.exists? %>
-          <% @order.all_adjustments.eligible.tax.group_by(&:label).each do |label, adjustments| %>
-            <tr>
-              <td></td>
-              <td><%= Spree.t(:tax) %> <%= label %>:</td>
-              <td><%= Spree::Money.new(adjustments.sum(&:amount), currency: @order.currency) %></td>
-            </tr>
-          <% end %>
-        <% end %>
-        <% @order.adjustments.eligible.each do |adjustment| %>
-          <% next if (adjustment.source_type == 'Spree::TaxRate') and (adjustment.amount == 0) %>
-          <tr>
-            <td></td>
-            <td><%= adjustment.label %>:</td>
-            <td><%= adjustment.display_amount %></td>
-          </tr>
-        <% end %>
-        <tr>
-          <td></td>
-          <td>
-            <%= Spree.t('order_mailer.confirm_email.total') %>
-          </td>
-          <td>
-            <%= @order.display_total %>
+          <td class="wrapper last">
+            <table class="twelve columns">
+              <%= render collection: @order.line_items, partial: 'spree/order_mailer/line_item' %>
+              <%= render 'spree/order_mailer/subtotal', order: @order %>
+              <% if @order.line_item_adjustments.exists? %>
+                <% if @order.all_adjustments.promotion.eligible.exists? %>
+                  <% @order.all_adjustments.promotion.eligible.group_by(&:label).each do |label, adjustments| %>
+                    <tr>
+                      <td class="six sub-columns">
+                        <%= Spree.t(:promotion) %> <%= label %>:
+                      </td>
+                      <td class="six sub-columns last right">
+                        <%= Spree::Money.new(adjustments.sum(&:amount), currency: @order.currency) %>
+                      </td>
+                    </tr>
+                  <% end %>
+                <% end %>
+              <% end %>
+              <% @order.shipments.group_by { |s| s.selected_shipping_rate.try(:name) }.each do |name, shipments| %>
+                <tr>
+                  <td class="six sub-columns">
+                    <%= Spree.t(:shipping) %> <%= name %>:
+                  </td>
+                  <td class="six sub-columns last right">
+                    <%= Spree::Money.new(shipments.sum(&:discounted_cost), currency: @order.currency) %>
+                  </td>
+                </tr>
+              <% end %>
+              <% if @order.all_adjustments.eligible.tax.exists? %>
```

Comment:
@damianlegawiec I think we do not need conditionals here as the loop will only be executed if its not empty.